src/core/network/payload.ts

```
@@ -0,0 +1,209 @@
+// file based on https://github.com/grammyjs/grammY/blob/d340d97c080b3e1bb34a593a9a238d4b8ac79c0f/src/core/payload.ts (MIT license)
+import { ReadableStream } from 'node:stream/web'
+import { TextEncoder } from 'node:util'
+
+export class StreamFile {
+  constructor(
+    readonly stream: () => AsyncIterable<Uint8Array>,
+    readonly name: string
+  ) {}
+}
+
+export type InputFile = StreamFile //| Blob?
+
+// required by Deno's `fetch`
+export function itrToStream<T>(iterator: AsyncIterator<T>): ReadableStream<T> {
+  return new ReadableStream({
+    async pull(controller) {
+      const { value, done } = await iterator.next()
+      if (done) {
+        controller.close()
+      } else {
+        controller.enqueue(value)
+      }
+    },
+  })
+}
+
+/**
+ * Calls `JSON.stringify` but removes `null` values from objects before
+ * serialization.
+ *
+ * @throws {TypeError} when encountering `InputFile`
+ */
+export function str(value: unknown) {
+  return JSON.stringify(value, (_, v) => {
+    if (v instanceof StreamFile) throw new TypeError() // caught by `createPayload`
```

This is a funny way to do it; but if we're doing this, I'd prefer it throw with a custom error so it doesn't swallow other possible `JSON.stringify` errors with invalid payloads.